src/System.Net.HttpListener/src/System/Net/Windows/RequestContextBase.cs

```diff
@@ -116,7 +125,19 @@ protected void UnsetBlob()
 
         protected void SetBuffer(int size)
         {
-            _backingBuffer = size == 0 ? null : new byte[size];
+            if(_backingBuffer != IntPtr.Zero)
+            {
+                Marshal.FreeHGlobal(_backingBuffer);
+            }
+
+            _backingBuffer = size == 0 ? IntPtr.Zero : Marshal.AllocHGlobal(size);
+            _backingBufferLength = size;
+
+            // Zero out the contents of the buffer.
```

Comment:
The buffer needs to be zero'd, which is done by default for managed memory. I'd like to do this faster, but I don't know what the best way is.